go-controller/pkg/ovn/pods.go

```diff
@@ -133,6 +133,10 @@ func (oc *Controller) deleteLogicalPort(pod *kapi.Pod) {
 
 	podDesc := pod.Namespace + "/" + pod.Name
 	logrus.Infof("Deleting pod: %s", podDesc)
+	hostSubnet, exists := oc.logicalSwitchCache[pod.Spec.NodeName]
```

Comment:
@JacobTanenbaum I'd actually run the delete stuff even if the node wasn't allocated. Maybe ovnkube quite and something in the API changed, and now the node has a different label. So on restart ovnkube master wouldn't put the node in the logical switch cache, but we still want to clean up. Maybe.

Also we may need to update syncPods(), but let's do that later since it's tricky because we don't have node information at that point.